### PR TITLE
chore: update project dependencies and adjust Webpack configuration

### PR DESCRIPTION
**Pull request by pankaj28843**
- Update project dependencies to latest versions for improved security and performance
- Modify .gitignore to exclude package-lock.json
- Create .nvmrc file to specify Node.js version for the project
- Adjust Webpack devServer static path in webpack.config.js

I tested this on the latest macOS, Windows 11, and Ubuntu 22.04

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v21.6
```
Comment (miltonhultgren) [inline .nvmrc]: I would use the LTS version instead.

**Review by miltonhultgren — APPROVED** [review]
Looks good to me, consider using the LTS version in the `.nvmrc` instead

File: .nvmrc (hunk 1 above)
Comment (pankaj28843) [reply .nvmrc]: Me too, I added this to align with what's in package.json. Will update

**Review by miltonhultgren — APPROVED** [review]

**Review by ahmagdy — APPROVED** [review]

**Comment by pankaj28843** [comment]
Can someone with write access please merge this PR? 🙇🏻‍♂️

**Comment by miltonhultgren** [comment]
@ahmagdy Has I think, I don't 🤷🏼

**Comment by ahmagdy** [comment]
@pankaj28843 @miltonhultgren, you both have access now. Let me know if it doesn't work

**Comment by pankaj28843** [comment]
Thank you @ahmagdy, I was able to merge. 🙌🏻

**Pull request merged**